analyses/focal-cn-file-preparation/rna-expression-validation.R

```diff
@@ -0,0 +1,175 @@
+    long_expression <- expression_df %>%
+      tidyr::gather(biospecimen_id, expression_value, -gene_id) %>%
+      dplyr::distinct()
+    long_expression$gene_id <-
+      gsub(".*\\_", "", long_expression$gene_id)
+
+    # Annotate expression data with metadata
+    expression_metadata <- long_expression %>%
+      dplyr::inner_join(metadata,
+                        by = c("biospecimen_id" = "Kids_First_Biospecimen_ID")) %>%
+      dplyr::select(
+        biospecimen_id,
+        expression_value,
+        gene_id,
+        Kids_First_Participant_ID,
+        tumor_descriptor
+      ) %>%
+      dplyr::distinct()
+
+    # Merge Focal CN data.frame with RNA expression data.frame
+    combined_df <- copy_number_df %>%
+      dplyr::distinct() %>%
+      dplyr::inner_join(
+        expression_metadata,
+        by = c(
+          "Kids_First_Participant_ID",
+          "gene_symbol" = "gene_id",
+          "tumor_descriptor"
+        )
+      ) %>%
+      dplyr::rename(biospecimen_id_cn = biospecimen_id.x, 
+                    biospecimen_id_expression = biospecimen_id.y) %>%
+      dplyr::mutate(log_expression_value = (log2(expression_value) + 1)) %>%
+      dplyr::mutate(expression_class = dplyr::case_when(
+        expression_value == 0 ~ "0",
+        expression_value > 0 & expression_value <= 10 ~ "0-10",
+        expression_value > 10 & expression_value <= 100 ~ "10-100",
+        expression_value > 100 ~ ">100"))
+
+    # Save results
+    readr::write_tsv(combined_df, file.path(results_dir, filename))
+
+    # Subset data.frame for plotting
+    combined_subset_df <- combined_df %>%
+      dplyr::filter(!(expression_class == ">100"))
+  }
+
+#### Join data -----------------------------------------------------------------
+
+# Add metadata to focal CN data.frame
+cn_df_metadata <- cn_df %>%
+  dplyr::filter(label %in% c("Hom_Deletion", "Hem_Deletion")) %>%
+  dplyr::inner_join(metadata,
+                    by = c("biospecimen_id" = "Kids_First_Biospecimen_ID")) %>%
+  dplyr::select(
+    gene_symbol,
+    biospecimen_id,
+    label,
+    copy_number,
+    tumor_ploidy,
+    Kids_First_Participant_ID,
+    tumor_descriptor
+  )
+
+# Merge Focal CN data.frame with RNA expression data.frame using
+# `merge_expression` custom function
+rsem_combined_polyA_df <-
+  merge_expression(cn_df_metadata,
+                   rsem_expression_polyA,
+                   metadata,
+                   "annotated_cn_polyA.tsv.gz")
+
+rsem_combined_stranded_df <-
+  merge_expression(cn_df_metadata,
+                   rsem_expression_stranded,
+                   metadata,
+                   "annotated_cn_stranded.tsv.gz")
+
+#### Plot and Save -------------------------------------------------------------
+
+png(
+  file.path(plots_dir, "cn_loss_expression_polyA.png"),
+  width = 1500,
+  height = 900,
+  res = 120
+)
+ggplot2::ggplot(rsem_combined_polyA_df,
+                ggplot2::aes(x = label, fill = expression_class)) +
+  ggplot2::geom_bar()
```

Comment:
Since the counts between these are so different, we probably want to use proportions on the y axis. Also this will make them order from bottom to top, which I think makes logical sense.
```suggestion
  ggplot2::geom_bar(position =  position_fill(reverse = TRUE)) +
  ylab("Proportion of called genes")
```